### PR TITLE
Feature: system status

### PR DESCRIPTION
**Pull request by shamoon**
## Proposed change

Seems like a useful but hopefully unobtrusive little thing to have. Im not totally certain about some of the methods for retrieving environment etc. but I got expected results when testing myself. Would be good to check that, false-positive errors (or false negatives) would suck. 

No rush on this, happy to iterate.

Screenshot 2024-02-13 at 11 53 09 PM
Screenshot 2024-02-13 at 11 54 45 PM
Screenshot 2024-02-15 at 11 24 29 PM
Screenshot 2024-02-13 at 11 35 05 PM
Screenshot 2024-02-12 at 11 17 54 AM
Screenshot 2024-02-13 at 11 56 51 PM

Closes #5704 

## Type of change

- [ ] Bug fix: non-breaking change which fixes an issue.
- [x] New feature / Enhancement: non-breaking change which adds functionality. _Please read the important note above._
- [ ] Breaking change: fix or feature that would cause existing functionality to not work as expected.
- [ ] Documentation only.
- [ ] Other. Please explain:

## Checklist:

- [x] I have read & agree with the [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/main/CONTRIBUTING.md).
- [x] If applicable, I have included testing coverage for new code in this PR, for [backend](https://docs.paperless-ngx.com/development/#testing) and / or [front-end](https://docs.paperless-ngx.com/development/#testing-and-code-style) changes.
- [x] If applicable, I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
- [x] If applicable, I have checked that all tests pass, see [documentation](https://docs.paperless-ngx.com/development/#back-end-development).
- [x] I have run all `pre-commit` hooks, see [documentation](https://docs.paperless-ngx.com/development/#code-formatting-with-pre-commit-hooks).
- [ ] I have made corresponding changes to the documentation as needed.
- [x] I have checked my modifications for any breaking changes.

### CODE REVIEW

**Comment by shamoon** [comment]
No rush on this

**Comment by noseshimself** [comment]
If possible, please add the underlying file system, too.

**Comment by shamoon** [comment]
It displays the info for the `/media/` storage.

**Comment by shamoon** [comment]
Given most users are running inside docker I’m not sure how much of that is available to the python backend. In general the goal here is not to be a comprehensive stack inspector, just to alert the user of potential problems and the basics of their setup.

**Comment by Gendra13** [comment]
Would it make sense to include the status of the search index as well as the classification model?

**Comment by shamoon** [comment]
> Would it make sense to include the status of the search index as well as the classification model?

Great idea

Screenshot 2024-02-13 at 4 40 39 PM
Screenshot 2024-02-13 at 4 40 30 PM

File: src/documents/tests/test_api_status.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+class TestSystemStatus(APITestCase):
+    ENDPOINT = "/api/status/"
+
+    def setUp(self):
+        self.user = User.objects.create_superuser(
+            username="temp_admin",
+        )
+
+    def test_system_status(self):
```
Comment (stumpylog) [inline src/documents/tests/test_api_status.py:22]: Nit: The little comment for tests

**Review by stumpylog — CHANGES_REQUESTED** [review]

File: src/documents/views.py [hunk 2]
```diff
@@ -1523,3 +1532,132 @@ class CustomFieldViewSet(ModelViewSet):
...
+            ]
+        except Exception as e:  # pragma: no cover
+            applied_migrations = []
+            db_status = "ERROR"
+            logger.exception(f"System status error connecting to database: {e}")
+            db_error = "Error connecting to database, check logs for more detail."
+
+        media_stats = os.statvfs(settings.MEDIA_ROOT)
+
+        redis_url = settings._CELERY_REDIS_URL
```
[the remark this answers is not in the document]
Comment (shamoon) [reply src/documents/views.py:1573]: Take your word on that!

**Comment by shamoon** [comment]
Thanks stumpy. Yea my thing with this one is a little unsure how it will behave in some environments (looking at you nixOS), may need some iterating...

File: src/documents/views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (stumpylog) [reply src/documents/views.py]: It's confusing/annoying that celery decided to make their own URL scheme

**Review by stumpylog — APPROVED** [review]

**Pull request merged**